test/ex_machina/ecto_test.exs

```diff
@@ -90,9 +131,10 @@ defmodule ExMachina.EctoTest do
 
     user = ExMachina.Ecto.assoc(MyApp.EctoFactories, attrs, :user)
 
-    created_user = %{id: 3, name: "John Doe", admin: false}
+    vals = %{name: "John Doe", admin: false}
+    created_user = struct(MyApp.User, vals) |> Map.put(:id, 1)
     assert user == created_user
-    assert_received {:created, ^created_user}
+    assert_received {:created, %MyApp.User{}}
```

Comment:
Combined with change you described bellow (i.e. changing the order of adding `id` and sending model), this would work...

The rationale I originally had for sending model _before_ adding id was to be able to test what is passed to `Ecto.Repo`. In _changeset_ PR from which this was extracted, `Ecto.Changeset` was passed to `Repo.Insert`, while `Ecto.Model` was returned as `created_record`.

Currently, no preprocessing is done on `record` passed to `Repo.Insert`, so not much is to test here, so I am :+1: on this... It will not be a hassle to switch order again if something worthy of testing comes up in future.